client/src/commands/run.ts

```diff
@@ -1,28 +1,30 @@
 // Copyright © 2022, SAS Institute Inc., Cary, NC, USA. All Rights Reserved.
 // Licensed under SAS Code Extension Terms, available at Code_Extension_Agreement.pdf
 
-import {
-  OutputChannel,
-  ProgressLocation,
-  ViewColumn,
-  window,
-  workspace,
-} from "vscode";
+import { OutputChannel, ProgressLocation, ViewColumn, window, workspace, Range } from "vscode";
 import { appendLog } from "../LogViewer";
 import { setup, run as computeRun } from "../viya/compute";
 
 let outputChannel: OutputChannel;
 
-function getCode(outputHtml: boolean): string {
-  const code = window.activeTextEditor.document.getText();
+function getCode(outputHtml: boolean, selected = false): string {
+  const editor = window.activeTextEditor;
+  const doc = editor?.document;
+
+  if (!editor || !doc) {
+    return "";
+  }
+
+  const code = selected
+    ? doc.getText(new Range(editor.selection.start, editor.selection.end))
```

Comment:
doc.getText(editor.selection) should be enough

Reply:
yes, I have updated.